packages/react-ui-components/src/components/SchemaForm/Templates/ArrayFieldTemplate/ArrayFieldTemplate.tsx

```diff
@@ -0,0 +1,55 @@
+import { EActionButtonType, EComponentSize, EIconName } from '@kelvininc/ui-components';
+import { ArrayFieldTemplateItemType, ArrayFieldTemplateProps, getTemplate, getUiOptions } from '@rjsf/utils';
+import React from 'react';
+import { KvActionButtonIcon, KvActionButtonText } from '../../../stencil-generated';
+import styles from './ArrayFieldTemplate.module.scss';
+
+const ArrayFieldTemplate = ({ idSchema, uiSchema, schema, title, disabled, readonly, required, items, canAdd, registry, onAddClick }: ArrayFieldTemplateProps) => {
+	const uiOptions = getUiOptions(uiSchema);
+	const ArrayFieldTitleTemplate = getTemplate<'ArrayFieldTitleTemplate'>('ArrayFieldTitleTemplate', registry, uiOptions);
+	const ArrayFieldDescriptionTemplate = getTemplate<'ArrayFieldDescriptionTemplate'>('ArrayFieldDescriptionTemplate', registry, uiOptions);
+	const ArrayFieldItemTemplate = getTemplate<'ArrayFieldItemTemplate'>('ArrayFieldItemTemplate', registry, uiOptions);
+
+	return (
+		<div className={styles['array-field-template']}>
+			<div className={styles['array-field-container']}>
+				<ArrayFieldTitleTemplate idSchema={idSchema} title={title} schema={schema} uiSchema={uiSchema} required={required} registry={registry} />
+
+				<ArrayFieldDescriptionTemplate
+					idSchema={idSchema}
+					description={uiOptions.description || schema.description}
+					schema={schema}
+					uiSchema={uiSchema}
+					registry={registry}
+				/>
+
+				<div className={styles['array-item-list']} key={`array-item-list-${idSchema.$id}`}>
+					{items && items.map(({ key, ...itemProps }: ArrayFieldTemplateItemType) => <ArrayFieldItemTemplate key={key} {...itemProps} />)}
+					{canAdd && AddButton({ canAdd, disabled, readonly, uiSchema, onAddClick })}
+				</div>
+			</div>
+		</div>
+	);
+};
+
+const AddButton = ({ disabled, readonly, uiSchema, onAddClick }: Partial<ArrayFieldTemplateProps>) => {
```

Comment:
Move this to another file.
Let's keep one component per file unless it's really small.